include/z64camera.h

```diff
@@ -203,26 +203,26 @@ typedef enum {
 
 typedef enum {
     /* 0x00 */ CAM_MODE_NORMAL,
-    /* 0x01 */ CAM_MODE_TARGET, // "PARALLEL"
-    /* 0x02 */ CAM_MODE_FOLLOWTARGET, // "KEEPON"
+    /* 0x01 */ CAM_MODE_Z_PARALLEL, // Holding Z but with no target, keeps the camera aligned
+    /* 0x02 */ CAM_MODE_Z_TARGET_FRIENDLY,
     /* 0x03 */ CAM_MODE_TALK,
-    /* 0x04 */ CAM_MODE_BATTLE,
-    /* 0x05 */ CAM_MODE_CLIMB,
-    /* 0x06 */ CAM_MODE_FIRSTPERSON, // "SUBJECT"
-    /* 0x07 */ CAM_MODE_BOWARROW,
-    /* 0x08 */ CAM_MODE_BOWARROWZ,
-    /* 0x09 */ CAM_MODE_HOOKSHOT, // "FOOKSHOT"
-    /* 0x0A */ CAM_MODE_BOOMERANG,
-    /* 0x0B */ CAM_MODE_SLINGSHOT, // "PACHINCO"
-    /* 0x0C */ CAM_MODE_CLIMBZ,
-    /* 0x0D */ CAM_MODE_JUMP,
-    /* 0x0E */ CAM_MODE_HANG,
-    /* 0x0F */ CAM_MODE_HANGZ,
-    /* 0x10 */ CAM_MODE_FREEFALL,
-    /* 0x11 */ CAM_MODE_CHARGE,
-    /* 0x12 */ CAM_MODE_STILL,
```

Comment:
STILL always seemed like a good way of describing its behavior to me. at least in terms of rotations it can do.
if youve ever ran around with isg and noticed that the camera is a bit stuck, its cause its constantly in this mode

notably in MM this still-ness effect is gone when swinging your sword. So I guess my opinion on this name hinges on: does MM still use this cam mode for attacks? or did they change it to a different one entirely?

Reply:
I thought about adding isg to the comment. I'll leave that to your judgement though if I should add it.

Interesting point about MM, I'll investigate and come back to this

Reply:
I dont think isg is worth commenting on, just think its the easiest example for people to recognize its "stillness" behavior

Reply:
camera modes describe more the current action the player is doing, than how the camera behaves

(yet another camdata explanation ahead)
how the camera behaves depends on the cam setting + cam mode combination, so for this "attack" mode how the camera behaves depends on the current setting, being "still" is a property of the camera function and data it maps to

Reply:
if this exact mode is still used for sword swinging in mm, and they removed the stillness behavior from the implementation, then im on board 100%. but still would like to make sure

Reply:
(also STILL is literally the original name, so it is their intention)

Reply:
my worry is that it can be used in more than just attack scenarios, and tunnel visioning it to attacks because of how they ended up using it in the final game is a bit misleading

Reply:
wdym it can be used in more than just attack scenarios? the purpose of this mode is to be activated when Link swings his sword (outside z-parallel/target)

Reply:
MM:

It's set in 3 places:
1) Attacking just like OoT but with a few extra horse related possibilities: (`PLAYER_STATE3_8` is a meleeWeaponAttack-related stateFlag)
```c
} else if (((Player_MeleeWeaponAttack == this->actionFunc) &&
            (this->meleeWeaponAnimation >= PLAYER_MWA_FORWARD_SLASH_1H) &&
            (this->meleeWeaponAnimation <= PLAYER_MWA_ZORA_PUNCH_KICK)) ||
            (this->stateFlags3 & PLAYER_STATE3_8) ||
            ((Player_RideHorse == this->actionFunc) && (this->genericTimer == 0)) ||
            (Player_DismountHorse == this->actionFunc)) {
    camMode = CAM_MODE_STILL;
```

2) Knockback (identical to OoT):
```c
} else if (Player_StartKnockback == this->actionFunc) {
    camMode = CAM_MODE_STILL;
```

3) New to MM: Shielding?
```c
} else if ((this->stateFlags1 & PLAYER_STATE1_SHIELDING) && (this->transformation != PLAYER_FORM_FIERCE_DEITY)) {
    camMode = CAM_MODE_STILL;
```

Reply:
I dont fully agree that it has to be for attacking specifically just cause of how the mode ends up getting used. but I will not continue to argue since the name "STILL" doesnt even make sense in MM anymore, since it is not actually still

Reply:
Even in OoT, `STILL` is set during knockback, which isn't quite an attack (unless you interpret knockback as a result of something attacking Link)

Reply:
but also, knockback? that makes it not attack only anymore lol
unless you wanna stretch it and say the player was attacked

Reply:
Reverted to as it was previously, `STILL` for now